katana/native/projects/static/doProjectHandling.js

```diff
@@ -0,0 +1,735 @@
+		alert("Please Specify a Project Engineer  ");
+		return; 
+	}
+
+	
+	jsonProjectObject['Details']['Name'] = katana.$activeTab.find('#projectName').val();
+	jsonProjectObject['Details']['Title'] = katana.$activeTab.find('#projectTitle').val();
+	jsonProjectObject['Details']['Engineer'] = katana.$activeTab.find('#projectEngineer').val();
+	jsonProjectObject['Details']['State'] = katana.$activeTab.find('#projectState').val();
+	jsonProjectObject['Details']['Date'] = katana.$activeTab.find('#projectDate').val();
+	jsonProjectObject['Details']['default_onError']['@action'] = katana.$activeTab.find('#default_onError').val();
+	jsonProjectObject['Details']['default_onError']['@value'] = katana.$activeTab.find('#default_onError_goto').val();
+	jsonProjectObject['Details']['Datatype'] = katana.$activeTab.find('#projectDatatype').val();
+	jsonProjectObject['SaveToFile'] = katana.$activeTab.find('#my_file_to_save').val();
+	//
+	// Now walk the DOM ..
+	// Create dynamic ID values based on the Suite's location in the UI. 
+
+	// Note that if we implement drag and drop we'll have to re-index the entire 
+	// visual display to reflect the movements of the order of objects on display 
+	// That would require a refresh after a drop anyway. 
+	//;
+	var url = "./projects/getProjectDataBack";
+	var csrftoken = $("[name='csrfmiddlewaretoken']").val();
+
+	$.ajaxSetup({
+			function(xhr, settings) {
+            xhr.setRequestHeader("X-CSRFToken", csrftoken)
+    	}
+	});
+	
+	var topNode  = { 'Project' : jsonProjectObject};
+
+
+	$.ajax({
+	    url : url,
+	    type: "POST",
+	    data : { 
+	    	'json': JSON.stringify(topNode),
+	    	//'Project': ns,
+	    	'filetosave': katana.$activeTab.find('#filesavepath').text() + "/" + $('#my_file_to_save').val()
+	    	},
+	    headers: {'X-CSRFToken':csrftoken},
+    
+    success: function( data ){
+        alert("Saved "+katana.$activeTab.find('#filesavepath').text() + "/" + $('#my_file_to_save').val());
+    	}
+	});
+
+}
+
+//
+// This creates the table for viewing data in a sortable view. 
+// 
+function createSuitesTable(xdata) {
+	var items = []; 
+	items.push('<table id="suite_table_display" class="project_configuration_table striped" width="100%">');
+	items.push('<thead>');
+	items.push('<tr id="suiteRow"><th>Num</th><th/><th>Suite</th><th>Execute</th><th>OnError</th><th>Impact</th><th/></tr>');
+	items.push('</thead>');
+	items.push('<tbody>');
+
+	katana.$activeTab.find("#tableOfTestSuitesForProject").html("");
+	for (var s=0; s<Object.keys(xdata).length; s++ ) {
+		var oneSuite = xdata[s];
+		//console.log(xdata);
+		if (oneSuite == null) {
+			xdata[s] = {} ;
+			oneSuite = xdata[s];
+		}
+		//console.log(oneSuite);
+		fillSuiteDefaults(s,xdata);
+		//console.log(oneSuite);
+		//console.log(oneSuite['path']);
+		
+		items.push('<tr data-sid="'+s+'">');
+		items.push('<td>'+(parseInt(s)+1)+'</td>');
+		var tbid = "textTestSuiteFile-"+s+"-id"+getRandomID();
```

Comment:
a katana-click function already have a reference to the jquery object of this
no need to use random id